File: pkg/src/com/jackmeng/prismix/user/_lua.java
// Software created by Jack Meng (AKA exoad). Licensed by the included "LICENSE" file. If this file is not found, the project is fully copyrighted.

package com.jackmeng.prismix.user;

import org.luaj.vm2.Globals;
import org.luaj.vm2.LuaValue;
import org.luaj.vm2.lib.jse.JsePlatform;

import com.jackmeng.ansicolors.jm_Ansi;

import static com.jackmeng.prismix.jm_Prismix.*;

/**
 * Objects for handling scripting with LUA including interop for calling Java
 * functions.
 *
 * @author Jack Meng
 */
public final class _lua
{
  public static _lua instance;

  private _lua()
  {

  }

  public static _lua instance() // lazy loaded singleton
  {
    if (instance == null)
      instance = new _lua();
    return instance;
  }

  public static Globals G()
  {
    return JsePlatform.standardGlobals();
  }

  public static Globals d_G()
  {
    return JsePlatform.debugGlobals();
  }

  public static String exec(String luacode)
  {
    return d_G().load(luacode).call().toString();
  }

  public static String exec(String luacode, LuaValue e)
  {
    return d_G().load(luacode).call(e).toString();
  }

  public static String exec(String luacode, LuaValue e, LuaValue e1)
  {
    return d_G().load(luacode).call(e, e1).toString();
  }

  public static String exec(String luacode, LuaValue e, LuaValue e1, LuaValue e2)
  {
    return d_G().load(luacode).call(e, e1, e2).toString();
  }

  public static void simple_load(String locale)
  {
    if (locale.endsWith((".lua")))
    {
      log("LUA", "Loading Lua script: " + locale);
      LuaValue chunk = G().loadfile(locale);
      chunk.call();
    }
    else
    {
      log("LUA", jm_Ansi.make("Failed to load: " + locale).red_bg().white_fg().toString()
          + " because the expected file extension \".lua\" was not found!");
    }
  }

}

File: pkg/src/com/jackmeng/prismix/user/_lua.kt
// Software created by Jack Meng (AKA exoad). Licensed by the included "LICENSE" file. If this file is not found, the project is fully copyrighted.
package com.jackmeng.prismix.user

import com.jackmeng.prismix.user._lua
import org.luaj.vm2.Globals
import org.luaj.vm2.lib.jse.JsePlatform
import org.luaj.vm2.LuaValue
import com.jackmeng.prismix.jm_Prismix
import com.jackmeng.ansicolors.jm_Ansi

/**
 * Objects for handling scripting with LUA including interop for calling Java
 * functions.
 *
 * @author Jack Meng
 */
object _lua
{
	var instance:_lua?=null
	fun instance():_lua? // lazy loaded singleton
	{
		if (instance==null) instance=_lua
		return instance
	}
	
	fun G():Globals
	{
		return JsePlatform.standardGlobals()
	}
	
	fun d_G():Globals
	{
		return JsePlatform.debugGlobals()
	}
	
	fun exec(luacode:String?):String
	{
		return d_G().load(luacode).call().toString()
	}
	
	fun exec(luacode:String? , e:LuaValue?):String
	{
		return d_G().load(luacode).call(e).toString()
	}
	
	fun exec(luacode:String? , e:LuaValue? , e1:LuaValue?):String
	{
		return d_G().load(luacode).call(e , e1).toString()
	}
	
	fun exec(luacode:String? , e:LuaValue? , e1:LuaValue? , e2:LuaValue?):String
	{
		return d_G().load(luacode).call(e , e1 , e2).toString()
	}
	
	fun simple_load(locale:String)
	{
		if (locale.endsWith(".lua"))
		{
			jm_Prismix.log("LUA" , "Loading Lua script: $locale")
			val chunk=G().loadfile(locale)
			chunk.call()
		}
		else
		{
			jm_Prismix.log(
				"LUA" ,
				jm_Ansi.make("Failed to load: $locale").red_bg().white_fg()
					.toString()+" because the expected file extension \".lua\" was not found!"
			)
		}
	}
}